app/models/declaration.rb

```diff
@@ -104,6 +105,11 @@ class Declaration < ApplicationRecord
 
     before_transition from: :ineligible, do: :clear_ineligibility_reason
 
+    before_transition on: :mark_as_eligible do |declaration, transition|
```

Comment:
I think we should just have setting the transition.args as separate calls on the declaration before we `mark_as_eligible` or `mark_as_ineligible`; it's not very clear on what this actually does and I expect someone coming at it from the other end will be confused as well as there's no `mark_as_eligible(payment_statement:)` method